test/wanda_web/controllers/execution_controller_test.exs

```diff
@@ -67,6 +67,26 @@ defmodule WandaWeb.ExecutionControllerTest do
     end
   end
 
+  describe "get last execution by group id" do
+    test "should return the last execution", %{conn: conn} do
+      %{group_id: group_id} = 10 |> insert_list(:execution) |> List.last()
+
+      json =
+        conn
+        |> get("/api/checks/groups/#{group_id}/executions/last")
+        |> json_response(200)
+
+      api_spec = ApiSpec.spec()
+      assert_schema(json, "ExecutionResponse", api_spec)
+    end
+
+    test "should return a 404", %{conn: conn} do
+      assert_error_sent 404, fn ->
```

Comment:
Question. How does this even know that it should return 404?
Does elixir capture the specific error from Ecto and is able to know that is not found?

Reply:
`Ecto.NoResultsError` exception is caught by a Plug that displays a 404, this is provided by Phoenix.
In development you'll see the phoenix debug page for every exception including the 404,  in production you'll just get a 404 error view.